### ISSUE TITLE
Add more information to MGLShapeSource cluster properties

### ISSUE THREAD

**Issue by kopacabana** [mapbox/mapbox-gl-native#9514]
**Platform:iOS**
**Mapbox SDK version:3.5**

Hello,

_Sorry for my awful english :)_
I implement your example for clustering points in MapBox iOS SDK.
That's work great !

### Request 1
Is it possible to receive more informations in cluster.attributes ?
When user tap on cluster, I receive : 
`
"properties": {
    cluster = 1;
    "point_count" = 3;
}
` 
but this will be nice if we can receive coordinates bounds or/and zoom level. With these informations, when user tap on cluster, we can zoom and center map on points/clusters includes in tapped cluster.
`
"properties": {
    cluster = 1;
    "point_count" = 3;
    "north" = 47.398767;
    "south" = 42.3467;
    "east" = 5.245;
    "west" = 3.3456;
    "zoom_level" = 12.2345;
}
` 
Now it's possible to use code below in handleTap function  : 
`var sw = CLLocationCoordinate2D(latitude: south, longitude: west)`
`var ne = CLLocationCoordinate2D(latitude: north, longitude: east)`
`mapView.setVisibleCoordinateBounds(MGLCoordinateBounds(sw: sw, ne: ne), edgePadding: UIEdgeInsets.zero, animated: true)` 

### Request 2
Create a protocol like : 
`func mapView(_ mapView: MGLMapView, calloutViewFor feature: MGLFeature) -> MGLCalloutView?`
With this, it's possible to reuse MGLCalloutView like for annotations !

Thanks for your SDK, really great.

**Comment by kopacabana** [comment]
Hi,

Do you plan to add this feature ? If yes, do you have a date or a period ?
I must finish a project where a I need it.

Thanks a lot

**Comment by boundsj** [comment]
Hi @kopacabana 

I've tagged this issue with Android and Core because I think this ticket actually describes a feature request for the Core GL implementation and a dependency it uses called [Supercluster](https://github.com/mapbox/supercluster.hpp). My best guess is that such a feature is currently blocked on https://github.com/mapbox/supercluster.hpp/issues/7 (see the js implementation as a reference in https://github.com/mapbox/supercluster/pull/31).

We don't currently have any immediate plans that I know of to implement this feature in this project but this ticket will be updated when we do.

cc @kkaefer @mourner

**Comment by 1ec5** [comment]
The corresponding GL JS issue is mapbox/mapbox-gl-js#2412.

**Comment by tobrun** [comment]
Tracking added supercluster.hpp 0.3.0 in https://github.com/mapbox/mapbox-gl-native/issues/12441 and https://github.com/mapbox/mapbox-gl-native/issues/12442. This will bring getLeaves, getChildren and getUnclusterZoomLevel. This API will address the use-cases from OP. Closing in favour of previously mentioned issues.
